File: 03-Stack/tlswldus03/B-10828.py
import sys
input = sys.stdin.readline

n = int(input())
stack = list()

for _ in range(n):
  commandLine = input().split()
  command = commandLine[0]
  if(command == "push"):
    stack.append(int(commandLine[1]))
  elif(command == "pop"):
    if(stack):
      print(stack.pop())
    else:
      print(-1)
  elif(command == "size"):
    print(len(stack))
  elif(command == "empty"):
    if(stack):
      print(0)
    else:
      print(1)
  elif(command == "top"):
    if(stack):
      print(stack[-1])
    else:
      print(-1)
      

File: 03-Stack/yujin0419/B-10828.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

stack<int> inStack;

bool notExist() {
	if (inStack.empty()) return true;
	else return false;
}

void resultPush() {
	int x;
	cin >> x;
	inStack.push(x);
}

void resultPop() {
	if (notExist()) cout << "-1" << "\n";
	else {
		cout << inStack.top() << "\n";
		inStack.pop();
	}
}

void countSize() {
	cout << inStack.size() << "\n";
}

void isEmpty() {
	if (notExist()) cout << "1" << "\n";
	else cout << "0" << "\n";
}

void resultTop() {
	if (notExist()) cout << "-1" << "\n";
	else {
		cout << inStack.top() << "\n";
	}
}

int main() {
	int commandCount;
	string command;

	cin >> commandCount;

	for (int i = 0; i < commandCount; i++) {
		cin >> command;
		if (command == "push") resultPush();
		else if (command == "pop") resultPop();
		else if (command == "size") countSize();
		else if (command == "empty") isEmpty();
		else if (command == "top") resultTop();
	}

	return 0;
}